### ISSUE TITLE
Decide on format for MIxS URIs - namespace

### ISSUE THREAD

**Issue by ramonawalls** [GenomicsStandardsConsortium/mixs#233]
**Go to this comment for solution: https://github.com/GenomicsStandardsConsortium/mixs/issues/233#issuecomment-657666511**

Although JSON does not strictly require term urls, much of what people need to do with mixs does (e.g., use mixs in linked data, use mixs terms in ontologies).

We discusses this topic at the CIG hackathon in Vienna in May.
Options include:

Use obo foundry purls
Make gensc purls
Make gensc URLs that are not purls (e.g., gensc.org/ns)
Keep namespace for terms in TDWG

Comment from @cmungall: Also https://w3id.org/

Comment from @lschriml: The last time we discussed this at the board level, there was a lot of support for:
Make gensc purls
--> Has this been discussed further on the CIG calls ?

Copy of GenomicsStandardsConsortium/mixs-ng#3

**Comment by kmexter** [comment]
> 



> **Decision on 10 May 2022:** We will use numerical IDs for each package and checklists. See the numbers at LINK. LinkML also makes the products of all packages and checklists. Let's make new IDs for those by concatonating the checklist number first then the package number, separated by an underscore.
> 
> _(Old comment, no longer valid: MIGO packages and checklists (is there more than one?) would be in https://w3id.org/gensc/migo)_
> 
> **note:** We may need to reconstruct some of the names to be sure there are no white spaces. They should all have dashes or underscores.
> 
> and all terms (regardless of where they are used) would be in https://w3id.org/gensc/terms/
> 
> Term URIs should use the unique 7 digit string and follow this format: https://w3id.org/gensc/terms/MIXS_0000001

Wondering if there is light at the end of the tunnel for having resolvable pages at the end of the PIDs for the MIXS terms? (I am asking as an open science data manager who is putting together DNA-derived species datasets to publish in portals). FYI I tried to load https://w3id.org/gensc/terms/ but my browser tells me there is a problem with the security certificate and it won't open that page (e.g. "When Chrome tried to connect to gensc.org this time, the website sent back unusual and incorrect credentials. ")

**Comment by ramonawalls** [comment]
@kmexter please see issue #390 for resolving links to terms. I am hiding your comment here, since it belongs on #390.

**Comment by ramonawalls** [comment]
# Background

This comment is about optimizing a user’s experience of accessing information about MixS terms via the LinkML auto-generated documentation pages. 

Users shouldn’t encounter any broken links, and GSC should use namespaces that are clear and authoritative, but also provide flexibility for the future. 

Management of these namespaces will potentially involve
•	changing the name of our GitHub Organization
•	selecting a w3id namespace and editing our w3id redirection rules
•	editing the prefixes and expansions in the LinkML YAML files

Decisions regarding the URLs for MIxS resources were made in 2020, before we started using LinkML seriously. Our knowledge and understanding have changed since then, and the documentation of the decisions is not completely clear. @turbomam and @ramonawalls met to try to clarify what resources a given URL should redirect to (what content should be associated with a given URL), and we make a new proposal below.

We have two layers of URLS - the verbose ones assigned by GitHub Pages and the terser ones available from w3id. Additionally, the LinkML files declare prefixes and their expansions. The GitHub Pages URLs are generated automatically based on the name of our GitHub organization and MIxS repo. Ideally, the LinkML prefixes would expand to w3id URLs, which in turn would redirect to the GitHub Pages.

Currently, https://github.com/GenomicsStandardsConsortium/mixs/blob/main/model/schema/mixs.yaml (note that this URL uses the organization name at the time of writing) declares:

```
mixs.vocab: https://w3id.org/mixs/vocab/
MIXS: https://w3id.org/mixs/terms/
MIGS: https://w3id.org/mixs/migs/
```

The MIGS prefix is not yet assigned to any checklists or terms and is not currently required. If we wish to use the https://w3id.org/gensc/mixs namespace, then those `mixs.vocab` and `MIXS` prefixes are wrong. They were just corrected in a PR based on branch #531. In that PR, @turbomam (with the blessing of @ramonawalls) changed it so we only have a single prefix for all MIxS resources, whether they are standards, terms, checklists, etc. 

# Feb. 2023 Proposal

I propose a solution that supports having future namespaces for standards outside MIxS (e.g., for MIGO for global observatories) but is also easy to maintain and understand and works well with GitHub Pages and  LinkML:

## gensc/mixs namespace

The root namespace for MIxS would be 

`https://w3id.org/gensc/mixs/`

Per issue #533, we will change the name of our Github organization to `genomicstandardsconsortium`. Therefore, GitHub pages will by necessity have the prefix `https:// genomicstandardsconsortium.github.io/mixs/`, which means that 

`https://w3id.org/gensc/mixs/`

should resolve to

`https:// genomicstandardsconsortium.github.io/mixs/`  which is the home page for the MIxS standard.

The key difference from the current state is that “gensc” would be added to the w3id redirect. This follows the w3id recommended practice of using “https://w3id.org/$org/$project/ and allows for the future prospect of including other GSC standards or projects, such as MIGO (minimum information about a genomic observatory).

MIxS checklists and extensions would use the same prefix. On 10 May 2022 we decided that checklists and extensions and all of their combinations would have numerical IDs. These numerical IDs have been assigned in the [incomplete schemasheets branch](https://github.com/GenomicsStandardsConsortium/mixs/tree/issue-511-tested-schemasheets) but not in the current release yet. In the schemasheets branch, IDs for combinations of checklists and extensions are constructed by concatenating the checklist number first then the extension number, separated by an underscore.

Following this proposal, our LinkML yaml file should be edited to generate documentation pages for checklist, extensions, and combinations of the following format:

MIGS bacteria checklist: `https:// genomicstandardsconsortium.github.io/mixs/0010003` (curie is MIXS:0010003)

Soil extension (aka environmental package): `https:// genomicstandardsconsortium.github.io/mixs/0016012`  (curie is MIXS:0016012)

Combination of MIGS bacteria and soil extension: `https:// genomicstandardsconsortium.github.io/mixs/0010003_ 0016012`

The shorter URLs generated from w3id for those three pages would be:

```
`https://w3id.org/gensc/mixs/0010003`
`https://w3id.org/gensc/mixs/0016012`
`https://w3id.org/gensc/mixs/0010003_ 0016012`
```

Our original proposal was that all terms, regardless of whether they are used in MIxS or some other (putative) standard should come within the w3id namespace `https://w3id.org/gensc/terms/`, so that each individual term ID would be of the form `https://w3id.org/gensc/terms/$term_unique_string`.

However, github pages automatically generatea URLs with the name of the organization and the repository. Although we could indeed use w3id URLs that contained `gensc/terms`, in order to have them redirect to corresponding documentation URLs generated by LinkML and GitHub pages (i.e. `https://genomicstandardsconsortium/github.io/terms/xxxxxxx’) we would need to move the term pages to their own repository (`https://genomicstandardsconsortium/terms/’)). **I don’t think it is worth the effort.**

Per comments above, we expect the term sets for other standards that may arise to be quite different from MIxS (e.g., they would describe observatories, not samples or sequences), and they could still reuse/import terms from the MIxS namespace. Therefore, I think it is a more practical solution to simply use the MIxS namespace for MIxS terms. Under this proposal, URLs for terms (i.e. slots) would follow the same format as URLs for packages and extensions (i.e. classes).

For example, the w3id URL for geo_loc_name (curie MIXS:0000010) would be:

`https://w3id.org/gensc/mixs/0000010`

And the documentation generated by GitHub pages would be: 

`https://genomicstandardsconsortium/github.io/mixs/0000010`.

Note a change from the earlier decision, where terms used PURLs of the format https://w3id.org/gensc/terms/MIXS_0000001.  Under the new plan, PURLs would follow the format https://w3id.org/gensc/mixs/0000001. In other words, we would change ‘terms’ to ‘mixs’ and we would not include `MIXS_` in the prefix. I am open to whether or not we include `MIXS_`  in the URL for terms, but it seems unnecessary to me. 

_Example of an actual term page for geolocation name (geo_loc_name):_

https://genomicsstandardsconsortium.github.io/mixs/0000010/

or

http://w3id.org/gensc/mixs/terms/0000010

**Comment by ramonawalls** [comment]
**What needs to happen:**

**In LinkML**, we should use the following prefix expansion rule:

`MIXS: https://w3id.org/gensc/mixs/`

So autogenerated documentation will make URLs for all MIxS element of the form: 

`https://genomicstandardsconsortium.github.io/mixs/0000010`

for any element of the standard. These are the URLs that are embedded into the github pages.

**In w3id.org**, we need to register the prefix 

`https://w3id.org/gensc/mixs/` redirects to `https://genomicstandardsconsortium.github.io/mixs/`

and for geo_loc_name

`https://w3id.org/gensc/mixs/0000010` redirects to `https://genomicstandardsconsortium.github.io/mixs/0000010`

This is a one way redirect, in that the short URLs are not in the documentation. However, since they are shorter and redirectable, we can use these as the "official" URLs for MIxS elements in publications, etc.

**Comment by ramonawalls** [comment]
We still need to decide how to implement this. I think it should be in our schemasheets, rather than continuing to hand edit https://github.com/GenomicsStandardsConsortium/mixs/blob/main/gsctools/mixs_converter.py. We will document that process in a different issue.
